src/overlays/actors/ovl_En_Attack_Niw/z_en_attack_niw.c

```diff
@@ -27,23 +36,371 @@ const ActorInit En_Attack_Niw_InitVars = {
     (ActorFunc)EnAttackNiw_Update,
     (ActorFunc)EnAttackNiw_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/EnAttackNiw_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/EnAttackNiw_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 63536, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 0, ICHAIN_STOP),
+};
+
+void EnAttackNiw_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnAttackNiw* this = THIS;
+    f32 tmp;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 25.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06002530, &D_060000E8, this->limbDrawTable, this->transitionDrawTable, 16);
+    if (this->actor.params < 0) {
+        this->actor.params = 0;
+    }
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.gravity = 0.0f;
+    this->unk_298.x = Math_Rand_CenteredFloat(100.0f);
+    this->unk_298.y = Math_Rand_CenteredFloat(10.0f);
+    this->unk_298.z = Math_Rand_CenteredFloat(100.0f);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.flags &= -2;
+    this->actor.shape.rot.y = this->actor.posRot.rot.y = (Math_Rand_ZeroOne() - 0.5f) * 60000.0f;
+    this->actionFunc = func_809B5670;
+}
+
+void EnAttackNiw_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAttackNiw* this = THIS;
+    EnAttackNiw* attached = (EnAttackNiw*)this->actor.attachedA;
+
+    if ((this->actor.attachedA != NULL) && (this->actor.attachedA->update != NULL) && (attached->unk_296 > 0)) {
+        attached->unk_296--;
+    }
+}
```

Comment:
I think it usually makes sense to write a null check without && since you want to first check if its not null before doing anything. same goes for the assignment to the temp (though that doesnt always work out)
Clearly a temp was needed to match, so at that point imo use it everywhere to make it look like something they would have written
```suggestion
void EnAttackNiw_Destroy(Actor* thisx, GlobalContext* globalCtx) {
    EnAttackNiw* this = THIS;
    EnAttackNiw* cucco; 

    if ((this->actor.attachedA != NULL) {
        cucco = (EnAttackNiw*)this->actor.attachedA;
        if (cucco->actor.update != NULL) && (cucco->unk_296 > 0)) {
            attached->unk_296--;
        }
    }
}
```
if the cucco assignment doesnt work in the if block then keep it up top, but thats what makes most sense imo

Reply:
Also, didnt we find out that attack cuccos attach to normal cuccos?